fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -124,6 +129,16 @@ public SystemInfoService() {
         pathHashToDishInfoRef = ImmutableMap.of();
 
         nodeSelector = new NodeSelector(this);
+        tabletComputeNodeMapper = new TabletComputeNodeMapper();
+    }
+
+    public boolean shouldUseInternalTabletToCnMapper() {
+        // We prefer to use the TabletComputeNodeMapper rather than delegating to StarOS/StarMgr for tablet->CN mappings
+        // if and only if we're using resource isolation groups.
+        return tabletComputeNodeMapper.numResourceIsolationGroups() > 1;
```

Comment:
does this mean that if say we have two non-default groups "group1" and "group2", and all compute nodes in "group2" are dropped for some reason (maybe a bug), then this function will evaluate to false, and all the resource isolation mechanisms will be bypassed entirely, and FEs in "group2" could use the compute nodes in "group1"?

Reply:
Thanks, looks good! I would also add a test case specifically to test this behavior (e.g. have just one group that is a non-default, and make sure the resource isolation mechanism is still used).

Reply:
Good point, done.